### PR TITLE
fixed tests

### LINKED ISSUE #68: Failure in joint_pos_controller.test
In R_hydro branch the test fails because the transmissions cannot find the actuators. The names with upper-case letters may be to blame.
The test is in package sr_mechanism_controllers.
The similar joint_pos_controller_with_fc.test also fails for the same reason

### PR DESCRIPTION
**Pull request by shadowmanos**
quick fix as there will be a new release pretty soon

closed #68 #72

### CODE REVIEW

File: sr_mechanism_controllers/include/sr_mechanism_controllers/test/test_controllers.hpp [hunk 1]
```diff
@@ -40,20 +38,26 @@ class TestControllers
...
-  boost::shared_ptr<pr2_mechanism_model::RobotState> robot_state;
-  boost::shared_ptr<TiXmlDocument> model;
-  boost::shared_ptr<sr_actuator::SrMotorActuator> actuator;
-  pr2_mechanism_model::JointState* joint_state;
+  controller::SrController *controller;
+  hardware_interface::HardwareInterface *hw;
+  ros_ethercat_model::RobotState *robot_state;
+  TiXmlDocument *model;
+  sr_mechanism_model::SimpleTransmission transmission;
+  ros_ethercat_model::JointState* joint_state;
```
Comment (shadowmanos) [inline sr_mechanism_controllers/include/sr_mechanism_controllers/test/test_controllers.hpp:60]: I was getting some weird errors about free() after the test had actually finished successfully. They were resolved by quiclky making everything old school pointers.

Comment (ugocupcic) [inline sr_mechanism_controllers/include/sr_mechanism_controllers/test/test_controllers.hpp:60]: Aren't we leaking memory now? (pointers but no free / delete)

Comment (shadowmanos) [inline sr_mechanism_controllers/include/sr_mechanism_controllers/test/test_controllers.hpp:60]: No we are not
lines 41-47 in header file

Comment (ugocupcic) [inline sr_mechanism_controllers/include/sr_mechanism_controllers/test/test_controllers.hpp:60]: Your headers are killing me! Thanks

**Pull request merged**